tools/docs/test/index.test.mjs

```diff
@@ -0,0 +1,78 @@
+import test from 'node:test';
+import fs from 'fs-extra';
+import { glob } from 'glob';
+import assert from 'node:assert';
+import MarkdownIt from 'markdown-it';
+import remark from 'remark';
+import path from 'node:path';
+
+const markdown = new MarkdownIt('zero');
+markdown.enable(['fence']);
+
+const root = path.resolve('tmp/docs');
+
+/**
+ * @param {any} node
+ * @param {Set<string>} files
+ * @param {string} file
+ */
+function checkNode(node, files, file) {
+  if (node.type === 'link') {
+    /** @type {import('mdast').Link} */
+    const link = node;
+    assert.ok(
+      !link.url.startsWith('/'),
+      'Link should be external or relative: ' + link.url,
+    );
+
+    if (link.url.startsWith('.')) {
+      const abs = path.resolve(
+        'tmp/docs',
+        path.dirname(file),
+        link.url.replace(/#.*/, ''),
+      );
+      // relative url
+      const target = abs.substring(root.length + 1);
+
+      assert.ok(
+        files.has(target),
+        `File not found: ${link.url} in ${file} -> ${target}`,
+      );
+    } else {
+      assert.ok(
+        !link.url.startsWith('https://docs.renovatebot.com/'),
+        'Docs links should be relative: ' + link.url,
+      );
+    }
+  } else if ('children' in node) {
+    for (const child of node.children) {
+      checkNode(child, files, file);
+    }
+  }
+}
+
+test('index', async (t) => {
+  await t.test('validate links', async (t) => {
+    const todo = await glob('**/*.md', { cwd: 'tmp/docs' });
+    const files = new Set(todo);
+
+    // add generated files
+    files.add('index.md');
+    files.add('release-notes-for-major-versions.md');
+    files.add('merge-confidence.md');
```

Comment:
When we add a new generated file, do we need to add it to the list here? If yes, we're probably going to forget to add the file to the test.

Could the docs test give a warning if a generated file is not listed in the test?

Reply:
it's only needed if there is a new file in docs repo and we've a link to it. the test with the link will fail if not added here. it logs the missing filename 😉

Reply:
The [`merge-confidence.md` file is in the main Renovate repository](https://github.com/renovatebot/renovate/blob/main/docs/usage/merge-confidence.md) now. I think this PR moved it from `whitesource/merge-confidence` to the main Renovate repository:

- #23215

The [`whitesource/merge-confidence` repository](https://github.com/whitesource/merge-confidence) is deprecated now:

> # Repository Deprecation Notice
> 
> **Deprecation Notice:** Although still supported, the method specified in the rest of this `README.md` file for enabling Merge Confidence badges is no longer the preferred approach.
> 
> For up-to-date instructions, we highly recommend referring to the official [documentation](https://docs.renovatebot.com/merge-confidence/).

I think this means you can drop the `merge-confidence.md` file from the generated files list?

Reply:
see other comments

Reply:
```suggestion
```

Reply:
The [`release-notes-for-major-versions.md` are in the docs repository](https://github.com/renovatebot/renovatebot.github.io/blob/main/src/release-notes-for-major-versions.md).

Should the release notes stay in the "generated files list"?